### PR TITLE
fix(Product CRUD): CRUD operation of product

### PR DESCRIPTION
**Pull request by TuyishimireEric**
#### What does this PR do?

CRUD operation of a product
#### Description of Task to be completed?

the user should be able to add/create items for sale
Such that the stock can be seen by any requesting client

#### How should this be manually tested?

- run `npm install` 
- run `npm run migrate`
- run `npm run seed`
- run `npm run dev`
- check the [swagger documentation ](http://localhost:3000/swagger/)

#### Any background context you want to provide?

-dummy data are provided before having the real data

#### What are the relevant pivotal tracker/Trello stories?

- id of the project #187300155

### CODE REVIEW

File: src/database/migrations/20240414205718-create-category.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        autoIncrement: true,
+        primaryKey: true,
+        type: Sequelize.INTEGER,
+      },
+      name: {
+        type: Sequelize.STRING,
+      },
+      createdAt: {
+        allowNull: false,
+        type: Sequelize.DATE,
```
Comment (fab-ryan) [inline src/database/migrations/20240414205718-create-category.js]: add status of category

**Review by Nadinefiona — APPROVED** [review]

File: src/config/config.js [hunk 2]
```diff
@@ -4,7 +4,7 @@ const dotenv = require('dotenv');
 dotenv.config();
 
 const getPrefix = () => {
-  const env = process.env.NODE_ENV || 'development';
+  const env = process.env.NODE_ENV || 'production';
```
Comment (fab-ryan) [inline src/config/config.js]: please use development by default

File: src/controllers/categoryController.ts [hunk 3]
```diff
@@ -0,0 +1,81 @@
+import { Request, Response } from 'express';
+import { CategoryAttributes } from '../database/models/category';
```
Comment (fab-ryan) [inline src/controllers/categoryController.ts:3]: you are missing update

**Review by fab-ryan — CHANGES_REQUESTED** [review]
work to those comments

File: src/config/config.js (hunk 2 above)
Comment (TuyishimireEric) [reply src/config/config.js]: I have changed thanks

File: src/controllers/categoryController.ts (hunk 3 above)
Comment (TuyishimireEric) [reply src/controllers/categoryController.ts:3]: Let me add it too

**Review by fab-ryan — APPROVED** [review]

**Review by musabehonore — APPROVED** [review]

**Review by Mukunzijames — APPROVED** [review]

**Pull request merged**